lib/rrdSupport.py

```diff
@@ -464,7 +464,10 @@ def fetch_rrd(self, filename, CF, resolution=None, start=None, end=None, daemon=
             args.append("--daemon")
             args.append(str(daemon))
 
-        return self.rrd_obj.fetch(*args)
+        if os.path.exists(filename):
+            return self.rrd_obj.fetch(*args)
+        else:
+            raise FileNotFoundError(f"File {filename} does not exist. Skipping...")
```

Comment:
Even if FileNotFoundError is more correct, I'd raise  RuntimeError which has been normally used to indicate problems w/ something in the library. Or check what is caught upstream.
```
            raise RuntimeError(f"RRD File '{filename}' does not exist. Failing fetch_rrd.")
```

Reply:
Checking to compare against the behavior in EL7 was a great suggestion and an eye-opening one! 

As per your suggestion, I used the code from the `rrdtool_crash.py` test script that was written during the troublehsooting of the issue #338 and executed the Python code in interpreter mode. Following is the output on EL7 and Alma9 platforms, respectively:
```
# EL7 - running rrdtool.x86_64 1.8.0-1.2.osg36.el7
>>> import rrdtool
>>> rrdo = rrdtool
# intentionally used the wrong file name to mimic the behavior of a non-existent file
>>> static_args = ['/var/lib/gwms-factory/work-dir/monitor/entry_fc445_ce-new/frontend_vofrontend_service_frontend/Stauts_Attributes.rrd', 'AVERAGE', '-r', '300', '-e', '1693520700', '-s', '1693513500']
>>> result = rrdo.fetch(*static_args)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
rrdtool.OperationalError: opening '/var/lib/gwms-factory/work-dir/monitor/entry_ITB_CE_EL9_SciToken/frontend_vofrontend_service_frontend/Stauts_Attributes.rrd': No such file or directory
```

```
# EL9 - running rrdtool.x86_64 1.7.2-21.el9
>>> import rrdtool
>>> rrdo = rrdtool
>>> static_args = ['/var/lib/gwms-factory/work-dir/monitor/entry_fc445_ce-new/frontend_vofrontend_service_frontend/Status_Attributes.rrd', 'AVERAGE', '-r', '300', '-e', '1693520700', '-s', '1693513500']
>>> result = rrdo.fetch(*static_args)
double free or corruption (out)
Aborted (core dumped)
```

With v1.8 coming from `osg36` repo on EL7, rrdtool throws a completely specific exception `rrdtool.OperationalError` when there is a non-existent file, which does not seem to be crashing the Python interpreter. On the other hand, with v1.7 coming from `appstream` repo on Alma9, the exception thrown appears to be crashing the Python interpreter level with the message `double free or corruption (out)`. Additionally, when the `rrdtool_crash.py` was run at the command line as `python rrdtool_crash.py`, the message returned was `Segmentation fault (core dumped)` which basically implied that the Python interpreter crashed.

@mambelli, Do you think using `OperationalError` to raise the error instead of RuntimeError would be better here?

Reply:
I still prefer RuntimeError. Or if you really want I'm OK with FileNotFoundError (checking the callers), at least these are both standard python exceptions and a bit more limited than Exception.
Both `rrdtool.OperationalError` (errors generated by rrdtool itself) and `rrdtool.ProgrammingError` (programming errors, e.g. passing wrong datatypes) are derived directly from `Exception` and custom to rrdtool.
I'd be for leaving these inside the rrdSupport.py module so that rrdtool does not have to be imported in other places.
If desired the rrdtool exceptions could be catched and chained to another raised exception.

Reply:
This has been taken into consideration and the latest commit addresses this.